### ISSUE TITLE
Clarification on Ego4D Dataset Usage and Selective Download

### ISSUE THREAD

**Issue by 10-OASIS-01** [ChenYi99/EgoPlan#16]
Hi, thanks for sharing this excellent work!

I have a question about the dataset:
In the documentation, it is mentioned to download the videos of Ego4D, and the folder structure is shown. Could you clarify:

1. Did you use the **entire Ego4D dataset** in your experiments, or only a subset?
2. If a subset is sufficient, could you provide instructions or metadata on how to **selectively download** the required videos (instead of the full dataset, which is very large)?

Thanks a lot!

**Comment by 10-OASIS-01** [comment]
Thanks a lot for the quick and helpful reply! 

The problem is now solved — for the dataset, one can directly download the pre-clipped Ego4D data from this repo: [[TencentARC/SEED-Bench-R1](https://github.com/TencentARC/SEED-Bench-R1)](https://github.com/TencentARC/SEED-Bench-R1).

Closing this issue, and really appreciate your support!

**Comment by Zeeyc** [comment]
I found that the `video_process` function in `*_interface.py` needs to be modified to adapt to the pre-cropped dataset. For instance, the following is my modified `ego4d_video_process1`.
```
def ego4d_video_process1(sample, vis_processor, n_frms=8):
    task_progress_metadata = sample["task_progress_metadata"]

    if len(task_progress_metadata) > 0:
        start_frames = [meta["start_frame"] for meta in task_progress_metadata]
        min_start_frame = min(start_frames)
        max_stop_frame = sample["current_observation_frame"]

        video_id = sample["video_id"]
        new_video_name = f"{video_id}-{min_start_frame:010d}_{max_stop_frame:010d}.mp4"
        original_dir = os.path.dirname(sample["video_path"])
        video_path = os.path.join(original_dir, new_video_name)

        video = cv2.VideoCapture(video_path)

        frame_offset = min_start_frame
        current_observation_frame_idx = sample["current_observation_frame"] - frame_offset

        video.set(cv2.CAP_PROP_POS_FRAMES, current_observation_frame_idx)
        ret, image = video.read()
        image = cv2.cvtColor(image, cv2.COLOR_BGR2RGB)
        image = Image.fromarray(image).convert("RGB")

        start_frame_idx = task_progress_metadata[0]["start_frame"] - frame_offset
        end_frame_idx = current_observation_frame_idx
        frame_indices = np.arange(start_frame_idx, end_frame_idx,
                                  (end_frame_idx - start_frame_idx) / (n_frms - 1)).astype(int).tolist()

        clip = []
        for i in frame_indices:
            video.set(cv2.CAP_PROP_POS_FRAMES, i)
            ret, frame = video.read()
            if ret:
                frame = cv2.cvtColor(frame, cv2.COLOR_BGR2RGB)
                frame = Image.fromarray(frame).convert("RGB")
                clip.append(frame)
        clip = clip[:n_frms - 1]
        clip.append(image)

        video.release()
    else:
        video_id = sample["video_id"]
        current_observation_frame = sample["current_observation_frame"]
        image_name = f"{video_id}-{current_observation_frame:010d}.png"
        original_dir = os.path.dirname(sample["video_path"])
        image_dir = os.path.join(original_dir, "images/Ego4D")
        image_path = os.path.join(image_dir, image_name)
        image = Image.open(image_path).convert("RGB")
        clip = [image] * n_frms
    clip = np.stack(clip)  # T, H, W, C
    clip = torch.tensor(clip).float().permute(3, 0, 1, 2)  # C, T, H, W
    clip = vis_processor.transform_video(clip)
    return clip
```
I'm not sure if it's because of my incorrect modification method, but the accuracy_for_egoplan_video_llama is only 0.4942. Is this normal? Do you have a more appropriate modification method?

> Thanks a lot for the quick and helpful reply!
> 
> The problem is now solved — for the dataset, one can directly download the pre-clipped Ego4D data from this repo: [[TencentARC/SEED-Bench-R1](https://github.com/TencentARC/SEED-Bench-R1)](https://github.com/TencentARC/SEED-Bench-R1).
> 
> Closing this issue, and really appreciate your support!
